addons/addon-base-raas-ui/packages/base-raas-ui/src/models/data-sources/DataSourceAccountsStore.js

```diff
@@ -0,0 +1,120 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+import _ from 'lodash';
+import { values } from 'mobx';
+import { types } from 'mobx-state-tree';
+import { consolidateToMap } from '@aws-ee/base-ui/dist/helpers/utils';
+import { BaseStore } from '@aws-ee/base-ui/dist/models/BaseStore';
+
+import { getDataSourceAccounts, checkStudyReachability, checkAccountReachability } from '../../helpers/api';
+import { DataSourceAccount } from './DataSourceAccount';
+import { DataSourceAccountStore } from './DataSourceAccountStore';
+
+// ==================================================================
+// DataSourceAccountsStore
+// ==================================================================
+const DataSourceAccountsStore = BaseStore.named('DataSourceAccountsStore')
+  .props({
+    accounts: types.map(DataSourceAccount),
+    accountStores: types.map(DataSourceAccountStore),
+    tickPeriod: 10 * 60 * 1000, // 10 minutes
+  })
+
+  .actions(self => {
+    // save the base implementation of cleanup
+    const superCleanup = self.cleanup;
+
+    return {
+      async doLoad() {
+        const accounts = await getDataSourceAccounts();
+        self.runInAction(() => {
+          consolidateToMap(self.accounts, accounts, (exiting, newItem) => {
+            exiting.setDataSourceAccount(newItem);
+          });
+        });
+      },
+
+      addAccount(raw) {
+        const id = raw.id;
+        const previous = self.accounts.get(id);
+
+        if (!previous) {
+          self.accounts.put(raw);
+        } else {
+          previous.setDataSourceAccount(raw);
+        }
+      },
+
+      getAccountStore(accountId) {
+        let entry = self.accountStores.get(accountId);
+        if (!entry) {
+          // Lazily create the store
+          self.accountStores.set(accountId, DataSourceAccountStore.create({ accountId }));
+          entry = self.accountStores.get(accountId);
+        }
+
+        return entry;
+      },
+
+      getStudyStore({ accountId, studyId }) {
```

Comment:
Why do we want a convenient method here to get StudyStore directly? In an idea world, component trying to use study store should already has account store right?

Reply:
It depends on if the store is a root store.  Usually, a react component might only need an MST entity and it gets the store it needs from a root store.  A root store is any store that is accessible from the appContext.  This way we don't have to pass stores everywhere.  However,  I do see some pros and cons doing this.  Anyway,  I have a screenshot to explain what root stores are.

At the end, I decided to remove this method and pass the store to the React component per your suggestion :)

![RaaS](https://user-images.githubusercontent.com/589501/102664264-63a6d880-4150-11eb-9b31-df9481e7b84e.png)